src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -222,9 +228,7 @@ private void CacheResourceDependencies(ResourceTypeReference childResourceTypeRe
                 ImmutableDictionary.Create<ModuleDeclarationSyntax, DiagnosticBuilder.ErrorBuilderDelegate>(),
                 ImmutableHashSet<ModuleDeclarationSyntax>.Empty);
 
-            // We'll use default bicepconfig.json settings during SnippetsProvider creation to avoid errors during language service initialization.
-            // We don't do any validation in SnippetsProvider. So using default settings shouldn't be a problem.
-            Compilation compilation = new Compilation(namespaceProvider, sourceFileGrouping, configurationManager.GetBuiltInConfiguration(disableAnalyzers: true));
+            Compilation compilation = new Compilation(namespaceProvider, sourceFileGrouping, configuration, linterAnalyzer);
```

Comment:
>linterAnalyzer [](http://example.com/codeflow?start=108&length=14)

Do we need to run the linter rules on the snippets at all?

Reply:
@majastrz , no, we don't run linter rules. It used the configuration with disabled analyzers:
https://github.com/Azure/bicep/blob/543bad68a3936cd720fc7119fcd00ae86bf8d488/src/Bicep.LangServer/Snippets/SnippetsProvider.cs#L82

Reply:
I will go ahead and merge this PR. I hope I answered your question. If there is any pending action item, I can take care of it in a follow up PR.